examples/advanced-topics/IIR-FIR/nelder-mead-filter-optimization.py

```diff
@@ -0,0 +1,271 @@
+        scipy_feedback_filter = np.insert(
+            -feedback_filter, 0, np.array([1])
+        )  # SciPy uses a different notation.
+        corrected_signal = signal.lfilter(
+            feedforward_filter, scipy_feedback_filter, distorted_waveform
+        )
+    else:  # Use the OPX to get the real data (simulated in this case).
+        config = {
+            "version": 1,
+            "controllers": {
+                "con1": {
+                    "type": "opx1",
+                    "analog_outputs": {
+                        1: {
+                            "offset": +0.0,
+                            "filter": {
+                                "feedback": feedback_filter,
+                                "feedforward": feedforward_filter,
+                            },
+                        },
+                    },
+                    "analog_inputs": {
+                        1: {"offset": +0.0},
+                    },
+                },
+            },
+            "elements": {
+                "flux1": {
+                    "singleInput": {"port": ("con1", 1)},
+                    "outputs": {"output1": ("con1", 1)},
+                    "intermediate_frequency": 10,
+                    "operations": {
+                        "readoutOp": "readoutPulse",
+                    },
+                    "time_of_flight": tof,
+                    "smearing": 0,
+                },
+            },
+            "pulses": {
+                "readoutPulse": {
+                    "operation": "measure",
+                    "length": pulse_len,
+                    "waveforms": {"single": "const_wf"},
+                    "digital_marker": "ON",
+                    "integration_weights": {"x": "xWeights", "y": "yWeights"},
+                },
+            },
+            "waveforms": {
+                "const_wf": {"type": "arbitrary", "samples": distorted_waveform},
+            },
+            "digital_waveforms": {
+                "ON": {"samples": [(1, 0)]},
+            },
+            "integration_weights": {
+                "xWeights": {
+                    "cosine": [1.0] * (pulse_len // 4),
+                    "sine": [0.0] * (pulse_len // 4),
+                },
+                "yWeights": {
+                    "cosine": [0.0] * (pulse_len // 4),
+                    "sine": [1.0] * (pulse_len // 4),
+                },
+            },
+        }
+
+        job = qmm.simulate(
+            config,
+            filter_optimization,
+            SimulationConfig(
+                duration=150,
+                simulation_interface=LoopbackInterface(
+                    [("con1", 1, "con1", 1)], latency=200
+                ),
+            ),
+        )
+        job.result_handles.wait_for_all_values()
+        corrected_signal = -job.result_handles.adc.fetch_all() / 4096
+
+    corr = np.correlate(corrected_signal, waveform, "full") / (
```

Comment:
explain this non trivial step in comment, maybe just saying it's explained in the readme